frontend/src/ui/component/exercise_list.rs

```diff
@@ -202,61 +235,80 @@ pub fn view(model: &Model, loading: bool, data_model: &data::Model) -> Vec<Node<
                     })
             ],
         ],
-        div![
-            C!["table-container"],
-            C!["mt-2"],
-            table![
-                C!["table"],
-                C!["is-fullwidth"],
-                C!["is-hoverable"],
-                tbody![exercises.iter().map(|e| {
-                    tr![td![
-                        C!["is-flex"],
-                        C!["is-justify-content-space-between"],
-                        C!["has-text-link"],
-                        span![
-                            ev(Ev::Click, {
-                                let exercise_id = e.id;
-                                move |_| Msg::Selected(exercise_id)
-                            }),
-                            e.name.to_string(),
-                        ],
-                        p![
-                            C!["is-flex is-flex-wrap-nowrap"],
-                            if model.view_edit {
-                                a![
-                                    C!["icon"],
-                                    C!["mr-1"],
-                                    ev(Ev::Click, {
-                                        let exercise_id = e.id;
-                                        move |_| Msg::EditClicked(exercise_id)
-                                    }),
-                                    i![C!["fas fa-edit"]]
-                                ]
-                            } else {
-                                empty![]
-                            },
-                            if model.view_delete {
-                                a![
-                                    C!["icon"],
-                                    C!["ml-1"],
-                                    ev(Ev::Click, {
-                                        let exercise_id = e.id;
-                                        move |_| Msg::DeleteClicked(exercise_id)
-                                    }),
-                                    i![C!["fas fa-times"]]
-                                ]
-                            } else {
-                                empty![]
-                            }
-                        ]
-                    ]]
-                })],
-            ]
-        ],
+        view_exercises(model, &current_exercises),
+        IF![!previous_exercises.is_empty() => nodes!(
+                    div![
+                        C!["container"],
+                        C!["has-text-centered"],
+                        C![format!("mb-1")],
```

Comment:
To make the margin consistent with the heading "Archives" on the Routines page:

```suggestion
                        C![format!("mb-3")],
```

Note that the `mb-1` doesn't have any effect, as the top margin of the table (`mt-2`) is bigger. Margins are not added up.

Reply:
I wasn't aware of that - changed.